document-legacy/src/operation.rs

```diff
@@ -274,23 +274,375 @@ pub enum Operation {
+				34.hash(state);
+				layer_path.hash(state);
+				manipulator_group.hash(state);
+			}
+			PushFrontManipulatorGroup { layer_path, manipulator_group } => {
+				35.hash(state);
+				layer_path.hash(state);
+				manipulator_group.hash(state);
+			}
+			RemoveManipulatorGroup { layer_path, id } => {
+				36.hash(state);
+				layer_path.hash(state);
+				id.hash(state);
+			}
+			RemoveManipulatorPoint { layer_path, id, manipulator_type } => {
+				37.hash(state);
+				layer_path.hash(state);
+				id.hash(state);
+				manipulator_type.hash(state);
+			}
+			TransformLayerInScope { path, transform, scope } => {
+				38.hash(state);
+				path.hash(state);
+				transform.iter().for_each(|x| x.to_bits().hash(state));
+				scope.iter().for_each(|x| x.to_bits().hash(state));
+			}
+			SetLayerTransformInScope { path, transform, scope } => {
+				39.hash(state);
+				path.hash(state);
+				transform.iter().for_each(|x| x.to_bits().hash(state));
+				scope.iter().for_each(|x| x.to_bits().hash(state));
+			}
+			TransformLayerScaleAroundPivot { path, scale_factor } => {
+				40.hash(state);
+				path.hash(state);
+				let (x, y) = scale_factor;
+				[x, y].iter().for_each(|x| x.to_bits().hash(state));
+			}
+			SetLayerScaleAroundPivot { path, new_scale } => {
+				41.hash(state);
+				path.hash(state);
+				let (x, y) = new_scale;
+				[x, y].iter().for_each(|x| x.to_bits().hash(state));
+			}
+			SetLayerTransform { path, transform } => {
+				42.hash(state);
+				path.hash(state);
+				transform.iter().for_each(|x| x.to_bits().hash(state));
+			}
+			ToggleLayerVisibility { path } => {
+				43.hash(state);
+				path.hash(state);
+			}
+			SetLayerVisibility { path, visible } => {
+				44.hash(state);
+				path.hash(state);
+				visible.hash(state);
+			}
+			SetLayerName { path, name } => {
+				45.hash(state);
+				path.hash(state);
+				name.hash(state);
+			}
+			SetLayerPreserveAspect { layer_path, preserve_aspect } => {
+				46.hash(state);
+				layer_path.hash(state);
+				preserve_aspect.hash(state);
+			}
+			SetLayerBlendMode { path, blend_mode } => {
+				47.hash(state);
+				path.hash(state);
+				blend_mode.hash(state);
+			}
+			SetLayerOpacity { path, opacity } => {
+				48.hash(state);
+				path.hash(state);
+				opacity.to_bits().hash(state);
+			}
+			SetLayerStyle { path, style } => {
+				49.hash(state);
+				path.hash(state);
+				style.hash(state);
+			}
+			SetLayerFill { path, fill } => {
+				50.hash(state);
+				path.hash(state);
+				fill.hash(state);
+			}
+			SetLayerStroke { path, stroke } => {
+				51.hash(state);
+				path.hash(state);
+				stroke.hash(state);
+			}
+			SetManipulatorHandleMirroring { layer_path, id, mirror_angle } => {
+				52.hash(state);
+				layer_path.hash(state);
+				id.hash(state);
+				mirror_angle.hash(state);
+			}
+			SetSelectedHandleMirroring { layer_path, toggle_angle } => {
+				53.hash(state);
+				layer_path.hash(state);
+				toggle_angle.hash(state);
```

Comment:
The large amount of boilerplate is perhaps something we should file an issue for?

Reply:
I think for now I'll just remove that commit and put it into its own pr